vcd/resource_vcd_nsxv_dhcp_relay.go

```diff
@@ -0,0 +1,434 @@
+		},
+	},
+}
+
+func resourceVcdNsxvDhcpRelay() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdNsxvDhcpRelayCreate,
+		Read:   resourceVcdNsxvDhcpRelayRead,
+		Update: resourceVcdNsxvDhcpRelayUpdate,
+		Delete: resourceVcdNsxvDhcpRelayDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdNsxvDhcpRelayImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name for DHCP relay settings",
+			},
+			"ip_addresses": {
+				Optional:    true,
+				Type:        schema.TypeSet,
+				Description: "A set of IP address of DHCP servers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"domain_names": {
+				Optional:    true,
+				Type:        schema.TypeSet,
+				Description: "A set of IP domain names of DHCP servers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"ip_sets": {
+				Optional:    true,
+				Type:        schema.TypeSet,
+				Description: "A set of IP set names which consist DHCP servers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"relay_agent": {
+				Required: true,
+				MinItems: 1,
+				Type:     schema.TypeSet,
+				Elem:     relayAgentResource,
+			},
+		},
+	}
+}
+
+// resourceVcdNsxvDhcpRelayCreate sets up DHCP relay configuration as per supplied schema
+// configuration
+func resourceVcdNsxvDhcpRelayCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	dhcpRelayConfig, err := getDhcpRelayType(d, edgeGateway, vdc)
+	if err != nil {
+		return fmt.Errorf("could not process DHCP relay settings: %s", err)
+	}
+
+	_, err = edgeGateway.UpdateDhcpRelay(dhcpRelayConfig)
+	if err != nil {
+		return fmt.Errorf("unable to update DHCP relay settings for Edge Gateway %s: %s", edgeGateway.EdgeGateway.Name, err)
+	}
+
+	// This is not a real object but a settings property on Edge gateway - creating a fake composite
+	// ID
+	fakeId, err := getDhcpRelaySettingsId(edgeGateway)
```

Comment:
I don't like using `fake` as it says something bad. `generatedId`, `constructedId` or something similar would be suitable

Reply:
compositeId